app/src/main/java/org/wikipedia/createaccount/CreateAccountActivityViewModel.kt

```diff
@@ -0,0 +1,111 @@
+package org.wikipedia.createaccount
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.wikipedia.WikipediaApp
+import org.wikipedia.dataclient.Service
+import org.wikipedia.dataclient.ServiceFactory
+import org.wikipedia.util.StringUtil
+import org.wikipedia.util.log.L
+
+class CreateAccountActivityViewModel : ViewModel() {
+    private val _createAccountInfoState = MutableStateFlow(AccountInfoState())
+    val createAccountInfoState = _createAccountInfoState.asStateFlow()
+
+    private val _doCreateAccountState = MutableStateFlow(CreateAccountState())
+    val doCreateAccountState = _doCreateAccountState.asStateFlow()
+
+    private val _verifyUserNameState = MutableSharedFlow<UserNameState>()
+    val verifyUserNameState = _verifyUserNameState.asSharedFlow()
+
+    private var verifyUserNameJob: Job? = null
+
+    fun createAccountInfo() {
+        viewModelScope.launch(CoroutineExceptionHandler { _, throwable ->
+            _createAccountInfoState.value = AccountInfoState.Error(throwable)
+        }) {
+            val response = withContext(Dispatchers.IO) {
+                ServiceFactory.get(WikipediaApp.instance.wikiSite).getAuthManagerInfo()
+            }
```

Comment:
We can ignore the `withContext()` here. See #4970

Reply:
Got it, applied the change in c5385617f794c183ac27993cc2ac008def662515